reference/2_constructing_data/2_4_syncer_splitter/README.md

```diff
@@ -0,0 +1,194 @@
+# Using the flow/Syncer & Splitter nodes for data manipulation
+
+The ```flow/Syncer``` node in Cranq is a synchronization primitive, that combines the data content of it's input signals into a single output stucture. It can be construct both arrays & key-value pairs. 
+
+The syncer node features a dynamic input port set (spread port):
+- It's ```fields``` input takes an array, which determines the quantity and names of its input ports
+- The names of these input ports will determine whether the node yields an array, or a key-value pair collection
+
+For example:
+ - ```fields``` value ```[0,1,2]``` will result in ports ```0, 1, 2```, and will yield an array with 3 elements
+ - ```fields``` value ```["a","b","c"]``` will result in ports ```a, b, c```, and will yiled a key-value pair collection with 3 elements
+
+> **_More info:_**
+>
+> For application flow use cases of the syncer/splitter nodes, check out  **[Application flow - Synchronization](../../1_application_flow/1_3_synchronization/README.md)**
+
+## Example - Building arrays with values
+
+> **_Try out:_**
+>
+> Place node **tutorials/data_constr/syncer_array** or **[download .cranqj](cranqj/data_constr_syncer_array.cranqj)** 
+
+Let's take the schema created in [Example - Building a complex type](../2_2_builders/README.md), and construct a collection of employee records:
+
+![](images/2021-07-15-15-51-55.png)
+
+- Let's create 2 employee records, by channeling our test data into a ```data/dictionary/Builder/keys & values``` node, as illustrated in the [example](../2_2_builders/README.md) mentioned above:
+    ```json
+    # schema:
+    ["EmpID","Name","Dept","HireDate","Salary"]
+
+    # record_1:
+    [100,"Ted","Accounting","2020-11-08",1500]
+
+    # record_2:
+    [101,"Sue","Facilities","2019-02-13",1500]
+    ```
+- Merge them together with the flow/Syncer node
+  - Note, that we are using the array-building feature of the node in this example
+  - We use  ```[0, 1]``` as it's fields input
+
+
+### Sample output:
+
+```json
+[
+  {
+    "EmpID": 101,
+    "Name": "Sue",
+    "Dept": "Facilities",
+    "HireDate": "2019-02-13",
+    "Salary": 1500
+  },
+  {
+    "EmpID": 100,

```

Comment:
I expected record with EmpID: 100 to be the first item in the output array as it comes in the port <0> of the syncher.

Reply:
Verified the downloadable cranqj file and record_1 contains EmpID 101 not EmpID 100 as described here